src/Compilers/CSharp/Test/Syntax/Parsing/RecordParsing.cs

```diff
@@ -408,6 +409,91 @@ public void RecordParsingAmbiguities()
             EOF();
         }
 
+        [Fact, WorkItem(45538, "https://github.com/dotnet/roslyn/issues/45538")]
+        public void RecordParsing_ConstraintsAndSemiColon()
+        {
+            var tree = ParseTree("record R<T> where T : class;", options: TestOptions.RegularPreview);
+            tree.GetDiagnostics().Verify();
+
+            UsingNode((CSharpSyntaxNode)tree.GetRoot());
```

Comment:
>UsingNode((CSharpSyntaxNode)tree.GetRoot()); [](start = 12, length = 44)

Wouldn't ```UsingTree``` helper work? Why use a different pattern to verify diagnostic and the tree?